File: java/293. Flip Game.java
public class Solution {
    public List<String> generatePossibleNextMoves(String s) {
        List<String> result = new ArrayList<String>();
        for(int i = 1; i < s.length(); ++i){
            if(s.charAt(i-1) == '+' && s.charAt(i) == '+'){
                result.add(s.substring(0,i-1) + "--" + s.substring(i+1));
            }
        }
        return result;
    }
}

//只要能找出连续的“++”，加上前面的和后面的，就得到了一个结果。

File: cpp/293. Flip Game.cpp
class Solution {
public:
    vector<string> generatePossibleNextMoves(string s) {
        vector<string> res;
        if(s == "" || s.size()<2) return res;
        for(int i = 0; i < s.size()-1; ++i){
            if(s[i] == '+' && s[i+1] == '+'){
                string str = s.substr(0,i) + "--"+s.substr(i+2, s.size()-i-2);
                
                res.push_back(s);
            }
        }
        return res;
    }
};